tests/fixture/tmpnet/network.go

```diff
@@ -501,10 +506,20 @@ func (n *Network) EnsureNodeConfig(node *Node) error {
 
 	// Set fields including the network path
 	if len(n.Dir) > 0 {
-		node.Flags.SetDefaults(FlagsMap{
+		defaultFlags := FlagsMap{
 			config.GenesisFileKey:    n.getGenesisPath(),
 			config.ChainConfigDirKey: n.getChainConfigDir(),
-		})
+		}
+
+		// Only set the subnet dir if it exists or the node won't start.
+		subnetDir := n.getSubnetDir()
+		if _, err := os.Stat(subnetDir); err == nil {
+			defaultFlags[config.SubnetConfigDirKey] = subnetDir
+		} else if err != nil && !errors.Is(err, os.ErrNotExist) {
```

Comment:
We're already guaranteed that `err != nil` in this check right?
```suggestion
		} else if !errors.Is(err, os.ErrNotExist) {
```